### PR TITLE
Add additional user attributes from SANIS api to UserInfo

### PR DESCRIPTION
**Pull request by ingokuba**

### CODE REVIEW

File: user-info-api/pom.xml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <parent>
+    <artifactId>fwu-kc-extensions</artifactId>
+    <groupId>de.intension</groupId>
+    <version>${projectVersion}</version>
+  </parent>
+  <modelVersion>4.0.0</modelVersion>
+
+  <artifactId>user-info-api</artifactId>
+  <version>1.0.0</version>
```
Comment (ingokuba) [inline user-info-api/pom.xml:13]: redundant - overrides version of parent

File: user-info-api/pom.xml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+  <properties>
+    <maven.compiler.source>11</maven.compiler.source>
+    <maven.compiler.target>11</maven.compiler.target>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+  </properties>
```
Comment (ingokuba) [inline user-info-api/pom.xml:19]: redundant - Overrides properties of parent

File: pom.xml [hunk 3]
```diff
@@ -116,6 +124,34 @@
...
+        <artifactId>guava</artifactId>
+        <version>31.1-jre</version>
+      </dependency>
+      <!-- Lombok -->
+      <dependency>
+        <groupId>org.projectlombok</groupId>
+        <artifactId>lombok</artifactId>
+        <version>1.18.24</version>
+        <scope>provided</scope>
+      </dependency>
```
Comment (ingokuba) [inline pom.xml:154]: Do you really need lombok?
Java 14 records will be the shit :)

File: user-info-api/src/main/java/de/intension/api/json/HeimatOrganisation.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class HeimatOrganisation
+{
+
+    @JsonProperty(UserInfoAttributeNames.ID)
+    private String id;
+    @JsonProperty(UserInfoAttributeNames.NAME)
+    private String name;
+    @JsonProperty(UserInfoAttributeNames.BUNDESLAND)
+    private String bundesland;
```
Comment (ingokuba) [inline user-info-api/src/main/java/de/intension/api/json/HeimatOrganisation.java:22]: Aren't all these attributes already defined by the field name of the DTO?

File: user-info-provider/pom.xml [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+  </parent>
+  <modelVersion>4.0.0</modelVersion>
+
+  <artifactId>user-info-provider</artifactId>
+
+  <properties>
+    <maven.compiler.source>11</maven.compiler.source>
+    <maven.compiler.target>11</maven.compiler.target>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+  </properties>
```
Comment (ingokuba) [inline user-info-provider/pom.xml:18]: redundant

File: user-info-provider/src/main/java/de/intension/mapper/user/UserInfoHelper.java [hunk 6]
```diff
@@ -0,0 +1,446 @@
...
+import org.keycloak.representations.IDToken;
+import org.keycloak.utils.StringUtil;
+
+import com.google.common.hash.Hashing;
+
+import de.intension.api.UserInfoAttribute;
+import de.intension.api.enumerations.*;
+import de.intension.api.json.*;
+
+public class UserInfoHelper
```
Comment (ingokuba) [inline user-info-provider/src/main/java/de/intension/mapper/user/UserInfoHelper.java:23]: woah big class - todo later

File: user-info-requester/pom.xml [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+  </parent>
+  <modelVersion>4.0.0</modelVersion>
+
+  <artifactId>user-info-requester</artifactId>
+
+  <properties>
+    <maven.compiler.source>11</maven.compiler.source>
+    <maven.compiler.target>11</maven.compiler.target>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+  </properties>
```
Comment (ingokuba) [inline user-info-requester/pom.xml:18]: redundant

**Review by ingokuba — COMMENTED** [review]
First review without readmes and user-info-requester

File: user-info-api/pom.xml (hunk 1 above)
Comment (mkunz-intension) [reply user-info-api/pom.xml]: your right, i will remove it

File: user-info-api/pom.xml (hunk 2 above)
Comment (mkunz-intension) [reply user-info-api/pom.xml:18]: your right, i will remove it

File: pom.xml (hunk 3 above)
Comment (mkunz-intension) [reply pom.xml:154]: This project is using Java 11 and therefore i will keep this dependency

File: user-info-api/src/main/java/de/intension/api/json/HeimatOrganisation.java (hunk 4 above)
Comment (mkunz-intension) [reply user-info-api/src/main/java/de/intension/api/json/HeimatOrganisation.java:22]: Yes, but the single source of truth is "UserInfoAttributeNames.class" and changes should be automatically reflected in the generated JSON file (just to be sure)

File: user-info-provider/pom.xml (hunk 5 above)
Comment (mkunz-intension) [reply user-info-provider/pom.xml:18]: your right, i will remove it

File: user-info-provider/src/main/java/de/intension/mapper/user/UserInfoHelper.java (hunk 6 above)
Comment (mkunz-intension) [reply user-info-provider/src/main/java/de/intension/mapper/user/UserInfoHelper.java:23]: let's keep it for now until we have a "stable"-release

File: user-info-requester/pom.xml (hunk 7 above)
Comment (mkunz-intension) [reply user-info-requester/pom.xml:18]: your right, i will remove it

**Pull request closed without merging**